Inspect: Create troubleshooting dashboard from a panel

ryantxu opened the pull request:
Reproducing (and then testing!) visualization issues is often complicated by trying to create a simple example that can run outside the original environment and communicate the key environment variables.

This adds an initial attempt at a programmatically generated troubleshooting dashboard that can help resolve visualization issues.  Query issues would likely require a different path.

This PR adds an additional dropdown to the panel inspect widget that will create a troubleshooting dashboard:

![63004e1a179fd170301295](https://user-images.githubusercontent.com/705951/185726476-19aabec6-cf48-4413-910b-470472d764e1.gif)

TODO:
- [ ] The datasource depends on having devenv testdata installed... we should likely make this part of the builtin "-- grafana --" datasource
- [ ] The text panel showing Panel JSON should use: https://github.com/grafana/grafana/pull/53850
- [ ] Show annotation data directly using https://github.com/grafana/grafana/pull/53998

ryantxu committed:
export troubleshooting dashboard

ryantxu committed:
import hack

ryantxu committed:
troubleshooting

ryantxu committed:
add randomizer

ryantxu committed:
randomizer kinda working

ryantxu committed:
more randomizing

joshhunt commented on .betterer.results:
```diff
@@ -4898,6 +4898,10 @@ exports[`better eslint`] = {
       [0, 0, 0, "Unexpected any. Specify a different type.", "2"],
       [0, 0, 0, "Unexpected any. Specify a different type.", "3"]
     ],
+    "public/app/features/inspector/troubleshooting.ts:5381": [
+      [0, 0, 0, "Unexpected any. Specify a different type.", "0"],
+      [0, 0, 0, "Unexpected any. Specify a different type.", "1"]
+    ],
```
i know this is just a draft for now, but when you're run please be sure not to commit _new_ exceptions into this file. Our goal is to remove betterer when these get down to zero, which we can't do if we add new exceptions.

prefer accurately typing new code, or using `// eslint-ignore` comments if there's absolutely no way to do that.

ryantxu committed:
Merge remote-tracking branch 'origin/main' into create-debug-dashboard

ryantxu commented:
Potentially fits better in the "Share panel" modal?  

<img width="806" alt="image" src="https://user-images.githubusercontent.com/705951/186020229-454e4397-cf90-4fc5-a31c-28409b91fcc5.png">

ryantxu committed:
Merge remote-tracking branch 'origin/main' into create-debug-dashboard

ryantxu committed:
update dash

ryantxu committed:
Merge remote-tracking branch 'origin/main' into create-debug-dashboard

ryantxu committed:
update dash

ryantxu committed:
Merge remote-tracking branch 'origin/main' into create-debug-dashboard

ryantxu committed:
use code settings

ryantxu committed:
use code settings

ryantxu committed:
Merge remote-tracking branch 'origin/main' into create-debug-dashboard

ryantxu committed:
text panel migrations

ryantxu committed:
Merge remote-tracking branch 'origin/main' into create-debug-dashboard

ryantxu committed:
Preferences: Support setting any dashboard as home, not just the starred ones (#54258)

bohandley committed:
OpenTsdb: migrate annotations from angular to react (#53856)

* remove annotation angular dependency and migrate annotations

* add nullish coalescing operator

* remove annotations angular partial

* make annotation queries code and errors more readable

* handle the betterer issue with '^ Unexpected any. Specify a different type' at top of file

* run precommits

* run prettier

yuri-tceretian committed:
Chore: Fix cue loader to use absolute path (#54257)

jalevin committed:
public-dashboards: Add log statement when public dashboard enabled or disabled (#54133)

* refactor apis for consistent outputs
* add dashboardUid validation at API layer
* add check for empty dashboardUid on SavePublicDashboard
* remove public dashboard errors from models package.

Co-authored-by: Ezequiel Victorero <evictorero@gmail.com>

adela-almasan committed:
Geomap: Local color range for dimensions (#54348)

Co-authored-by: Ryan McKinley <ryantxu@gmail.com>

nmarrs committed:
Geomap: cleanup (#54328)

JohnnyQQQQ committed:
Alerting: add missing yaml tag to mute time struct (#54287)

JohnnyQQQQ committed:
Alerting: use raw query data in provisioning to bypass interpolation of macros (#54293)

ryantxu committed:
Storage: Avoid UID errors from annotations api when loading dashboards from storage (#54346)

ying-jeanne committed:
Chore: Use db.DB interface instead of sqlstore (#54358)

* use db.DB interface instead of sqlstore

* make store service depends on db.DB instead of sqlstore

orlandoortegajr committed:
Chore: Added controls to DatePickerWithInput story (#54360)

linoman committed:
Chore: Group auth docker blocks (#54274)

* Move auth docker images to parent folder

* Add README file for auth parent folder

* Add grafana-authnz-team as code owner

* Update `devenv` docs

mgyongyosi committed:
Docs: Fix typo in custom-role-actions-scopes.md (#54294)

* Fix typo in custom-role-actions-scopes.md

* Address lint issue

marefr committed:
Chore: Fix prometheus docker block (#54388)

Clarity-89 committed:
TimeRangeForm: Do not use HTML form (#54318)

* TimeRangeForm: Do not use HTML form

* TimeRangeForm: rename to TimeRangeContent

asimpson committed:
QueryEditorRow: filter data on mount (#54260)

This works correctly on initial dashboard load. However, when an action
is taken that triggers a "re-mount", e.g. opening the options pane on
the right side of the page, the `data` prop becomes undefined.

ivanortegaalba committed:
Analytics: Rename create dashboard event (#54390)

adela-almasan committed:
Geomap: Fix tooltip display (#54245)

linoman committed:
Update README build references (#54394)

mmandrus committed:
Secrets: Implement admin mechanism for deleting all secrets stored on the secrets plugin (#54264)

* implement delete all secrets endpoint

* change deletion check to just check for installed plugin

* refactor function call

MattDodsonEnglish committed:
Fix admonition (#54408)

evictorero committed:
Public Dashboards: use intervalMs and maxDataPoints from request (#53613)

adela-almasan committed:
Geomap: Cleanup (#54415)

ryantxu committed:
Merge remote-tracking branch 'origin/main' into create-debug-dashboard

ryantxu committed:
Merge remote-tracking branch 'origin/main' into create-debug-dashboard

ryantxu commented:
initial version merged with #54417
